### PR TITLE
Dependency Graph and Better Context

### PR DESCRIPTION
**Pull request by AbhijeetDash**
# Change Log
### Feature

- Added Support for Tree-sitter parsers for selected languages.
- Added new command structure for config command.
- Converted local and global to sub-commands with options.
- Local sub-command introduces a new parse option `overide config local --parse | -pa`.
- Added support for incrementally generating and updating the `oi-dependency.json`.
#### Supported Languages

Overide now supports creating dependency graph in the following languages. However testing needs to be done on all the languages to add cases for identifying the parsing node case.
  
```
C, C++, Java, Python, Ruby, Go, JavaScript, TypeScript, C#
```
### Enhancements

- Updated the `prompt.structure.json` and removed the `complete`  param.
- Updated `service.system.prompt` to better structure prompts using message roles.
- Restructured Network layer to remove redundancy and improve data flow in code.
- Updated `process.response` to remove redundancy.
### Bug Fixes.

- Fixed the way start command ignored files from `oi-config.json`
- Fixed bug in replacing the prompt with insertion code.

### Misc

Added `funding.json` to participate in [FLOSS Funding](https://dir.floss.fund/). This is a initiative by the Zerodha which is a leading investment firm in India. FLOSS would allow us to manage expenses related to building and testing the project.

### CODE REVIEW

File: .changeset/gorgeous-pugs-peel.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+"overide": minor
+---
+
+Adds: Support for dependency graph generation with tree sitter.
+Updated: 
+1. The Network layer is refactored to reduce redundancy.
+2. Config command now has two sub-commands 'local' & 'global' with respective   options.
```
Comment (Zoheb-Malik) [inline .changeset/gorgeous-pugs-peel.md]:
```suggestion
2. Config command now has two sub-commands 'local' & 'global' with respective options.
```

File: assets/prompt.structure.json
Comment (Zoheb-Malik) [inline assets/prompt.structure.json:1]: Looks fine for now, although eventually I would like to make a fix on the entire codebase that will lessen code duplication.

File: src/commands/command.config.ts [hunk 2]
```diff
@@ -191,37 +245,21 @@ class Config extends OiCommand {
           }
         }
       });
+
+      console.log('Following file will be watched :\n');
```
Comment (Zoheb-Malik) [inline src/commands/command.config.ts]:
```suggestion
      console.log('The following file will be watched:\n');
```

File: src/handlers/handler.start.ts [hunk 3]
```diff
@@ -108,59 +74,30 @@ class StartCommandHandlerImpl extends StartCommandHandler {
...
-        insertionRequest.fileContent,
-        insertionRequest.prompt
-      );
-      // Create a prompt context by extracting surrounding content
-      const promptArray = await userPromptService.findPromptContext(
-        insertionRequest.fileContent,
-        insertionRequest.prompt,
-        verbose
-      );
+      // TODO : Create prompt context using dependency graph.
```
Comment (Zoheb-Malik) [inline src/handlers/handler.start.ts:77]: Coming soon?

**Review by Zoheb-Malik — APPROVED** [review]
A lot of good things. Although I have approved, I did leave some comments for you

File: src/handlers/handler.start.ts (hunk 3 above)
Comment (AbhijeetDash) [reply src/handlers/handler.start.ts:77]: The embeddings part is still not done. Right now I used a graph ranking algorithm to find the most relevant node (file). A relevant node is basically a import statement to a file, this approach still sends a lot of code with prompts with embeddings we will be able to send only used part of every node. 
So I left this TODO to remind me from where to start the embeddings integrations.

File: assets/prompt.structure.json
Comment (AbhijeetDash) [reply assets/prompt.structure.json:1]: Sounds like a good plan. Let's do it incrementally. I see there is a lot of duplication in the insertion of code part as well, so I was aiming to fix that part with the introduction of embeddings.

**Pull request merged**